src/main/java/uk/gov/companieshouse/authcodenotification/service/PrivateDataRetrievalService.java

```diff
@@ -32,12 +32,18 @@ public OverseasEntityDataApi getOverseasEntityData(String requestId, String comp
                     .execute()
                     .getData();
 
-            ApiLogger.infoContext(requestId, "Successfully retrieved overseas entity data from database",  logDataMap.getLogMap());
+            ApiLogger.infoContext(requestId, "Successfully retrieved overseas entity data from database", logDataMap.getLogMap());
             return overseasEntityDataApi;
-        } catch (URIValidationException | IOException e) {
-            var message = "Error retrieving overseas entity data from database";
-            ApiLogger.errorContext(requestId, message, e, logDataMap.getLogMap());
-            throw new ServiceException(e.getMessage(), e);
+        } catch (URIValidationException | ApiErrorResponseException e) {
+            if (e instanceof ApiErrorResponseException && ((ApiErrorResponseException)e).getStatusCode() == 404) {
+                var message = "Http exception status: " + ((ApiErrorResponseException)e).getStatusCode();
+                ApiLogger.errorContext(requestId, message, e, logDataMap.getLogMap());
```

Comment:
might need to switch the error logging to info here, see what martin and phil say on the other related comment

Reply:
It feels that at this point this is an error (it's the controller code that kinds of 'downgrades' the severity and makes it a 404) so suggest leaving as error.

Reply:
Is this resolved?

Reply:
i was thinking for the 404 message, could we add something to indicate what it was doing when not found, eg "Unable to find overseas entity data from database, HTTP exception status " ...

And similar for the PublicDataRetrieval - "Unable to find company profile, HTTP exception status " ...

Reply:
Updated